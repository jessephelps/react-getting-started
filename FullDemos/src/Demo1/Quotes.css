.Quote {
  padding: 1em;
}