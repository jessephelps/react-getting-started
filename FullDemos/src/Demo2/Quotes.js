import React, { Component } from 'react';
import Quote from './Quote';

class Quotes extends Component {
  render() {
    return (
      <div className="Quotes">
        <Quote text={'I know you think you understand what you thought I said but I\'m not sure you realize that what you heard is not what I meant - Alan Greespan'} />
        <Quote text={'I guess I should warn you, if I turn out to be particularly clear, you\'ve probably misunderstood what I\'ve said - Alan Greenspan'} />
        <Quote text={'It\'s okay if you disagree with me. I can\'t force you to be right.'} />
      </div>
    );
  }
}

export default Quotes;
