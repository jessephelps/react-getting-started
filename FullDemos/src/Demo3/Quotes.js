import React, { Component } from 'react';
import Quote from './Quote';
// import Quote from './PureQuote';
// import Quote from './StatelessQuote';

const quoteList = [
  'Not everything that can be counted counts, and not everything that counts can be counted.',
  'A person who never made a mistake never tried anything new.',
  'I have not failed. I\'ve just found 10,000 ways that won\'t work.',
];

class Quotes extends Component {
  constructor() {
    super();
    this.state = { quoteIndex: 0 };
  }

  getNextQuoteIndex = (index) => {
    return index + 1;
  }

  goToNextQuote = () => {
    this.setState({ quoteIndex: this.getNextQuoteIndex(this.state.quoteIndex) });
  }

  getQuoteText = () => {
    return quoteList[this.state.quoteIndex];
  }

  render() {
    return (
      <div className="Quotes">
        <Quote text={this.getQuoteText()} />
        <button onClick={this.goToNextQuote}>Next</button>
      </div>
    );
  }
}

export default Quotes;
