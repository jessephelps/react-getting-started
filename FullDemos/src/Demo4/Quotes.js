import React, { Component } from 'react';
import Quote from './Quote';

class Quotes extends Component {
  constructor() {
    super();
    this.state = { quote: '', quoteIndex: 0 };
  }

  componentDidMount = () => {
    this.getNextQuote();
  }

  getNextQuote = async () => {
    let nextIndex = this.state.quoteIndex + 1;
    let newQuoteResponse = await fetch(`${nextIndex}.json`);
    if (newQuoteResponse.redirected) {
      nextIndex = 1;
      newQuoteResponse = await fetch(`${nextIndex}.json`);
    }
    const newQuoteResponseJson = await newQuoteResponse.json();
    this.setState({quote: newQuoteResponseJson.quote, quoteIndex: nextIndex});
  }

  render() {
    return (
      <div className="Quotes">
        <Quote text={this.state.quote} />
        <button onClick={this.getNextQuote}>Next</button>
      </div>
    );
  }
}

export default Quotes;
