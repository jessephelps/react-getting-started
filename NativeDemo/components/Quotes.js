import React, { PureComponent } from 'react';
import { StyleSheet, Text, View, Button } from 'react-native';

const Quote = (props) => (
  <View>
    <Text>
      {props.text}
    </Text>
  </View>
);

class Quotes extends PureComponent {
  render() {
    return (
      <View>
        <Quote text={this.props.quote} />
        <Button title={'Next'} onPress={this.props.getNextQuote} />
      </View>
    );
  }
}

export default Quotes;
