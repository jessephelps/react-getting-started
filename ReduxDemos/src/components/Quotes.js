import React, { Component } from 'react';
import Quote from '../components/Quote';

class Quotes extends Component {
  componentDidMount = () => {
    this.getNextQuote();
  }

  render() {
    return (
      <div className="Quotes">
        <Quote text={this.props.quote} />
        <button onClick={this.props.getNextQuote}>Next</button>
      </div>
    );
  }
}

export default Quotes;
