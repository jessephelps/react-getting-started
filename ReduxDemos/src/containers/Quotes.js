import React, { Component } from 'react';
import { connect } from 'react-redux';

import { getNextQuote } from '../actions/index';
import Quote from '../components/Quote';

class Quotes extends Component {
  componentDidMount = () => {
    this.props.getNextQuote();
  }

  render() {
    return (
      <div className="Quotes">
        <Quote text={this.props.quote} />
        <button onClick={this.props.getNextQuote}>Next</button>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    quote: state.quote,
  };
};

const mapDispatchToProps = dispatch => ({
  getNextQuote: () => {
    dispatch(getNextQuote());
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(Quotes);
