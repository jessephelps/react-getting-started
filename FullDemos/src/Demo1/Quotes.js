import React, { Component } from 'react';
import Quote1 from './Quote1';
import Quote2 from './Quote2';
import Quote3 from './Quote3';
import './Quotes.css';

class Quotes extends Component {
  render() {
    return (
      <div className="Quotes">
        <Quote1 />
        <Quote2 />
        <Quote3 />
      </div>
    );
  }
}

export default Quotes;
