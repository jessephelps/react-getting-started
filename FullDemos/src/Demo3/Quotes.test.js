import React from 'react';
import renderer from 'react-test-renderer';
import Quotes from './Quotes';

it('quote changes when the Next button is clicked', () => {
  // arrange
  
  // act
  const component = renderer.create(<Quotes />);
  let tree = component.toJSON();

  // assert
  expect(tree).toMatchSnapshot();

  // act
  tree.children[1].props.onClick();
  tree = component.toJSON();

  // assert
  expect(tree).toMatchSnapshot();
});
