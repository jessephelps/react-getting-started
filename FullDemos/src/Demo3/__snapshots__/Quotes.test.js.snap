// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`quote changes when the Next button is clicked 1`] = `
<div
  className="Quotes"
>
  <div
    className="Quote"
  >
    Not everything that can be counted counts, and not everything that counts can be counted.
  </div>
  <button
    onClick={[Function]}
  >
    Next
  </button>
</div>
`;

exports[`quote changes when the Next button is clicked 2`] = `
<div
  className="Quotes"
>
  <div
    className="Quote"
  >
    A person who never made a mistake never tried anything new.
  </div>
  <button
    onClick={[Function]}
  >
    Next
  </button>
</div>
`;
